### ISSUE TITLE
Debugger Display for property yields incorrect result

### ISSUE THREAD

**Issue by AlexanderMorou** [dotnet/roslyn#20010]
**Version Used**: 
15.2 (26430.12) Release

**Steps to Reproduce**:

```
class Program
{
    static void Main(string[] args)
    {
        ITest myTest = new Test2();
        Console.WriteLine(myTest.TestBool is string);
    }
}

public class Test1 :
    ITest
{
    public bool TestBool { get { return false; } }
}

public class Test2 :
    Test1,
    ITest2
{
    public new object TestBool { get { return "true"; } }
}

public interface ITest
{
    bool TestBool { get; }
}
public interface ITest2 :
    ITest
{
    new object TestBool { get; }
}

```
Set a breakpoint on `Console.WriteLine(myTest.TestBool is string);`

**Expected Behavior**:
Mousing over `myTest.TestBool`, within the `Console.WriteLine` call, should show 'false'.

**Actual Behavior**:
Mousing over `myTest.TestBool`, within the `Console.WriteLine` call, shows "true"

**Note**:
This can cause odd side-effects which make debugging accurately difficult, if the value you see in the debugger is incorrect.

**Comment by jcouv** [comment]
That seems odd indeed.

Here's what QuickWatch shows:
![image](https://cloud.githubusercontent.com/assets/12466233/26802056/eaddeee0-49f3-11e7-9345-5654aa4d1597.png)

![image](https://cloud.githubusercontent.com/assets/12466233/26802131/32cfc750-49f4-11e7-8784-ff149bc2aa2e.png)

**Comment by jcouv** [comment]
FYI @cston

**Comment by jcouv** [comment]
@cston do you know if there is a corresponding debugger issue tracking this that we could link? Should the present issue be assigned?

**Comment by jinujoseph** [comment]
tracked here https://devdiv.visualstudio.com/DevDiv/_workitems?id=452642&_a=edit @thanks @cston

**Comment by AlexanderMorou** [comment]
Just curious, but is this still on point for 15.5?

The link provided is inaccessible to me.

**Comment by jcouv** [comment]
@AlexanderMorou Yes, the linked issue was resolved as fixed for 15.5.

**Comment by AlexanderMorou** [comment]
Confirmed fixed in 15.5.1.  Thanks!
